server/src/main/java/com/hedera/block/server/persistence/storage/FileSystemBlockStorage.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.persistence.storage;
+
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import io.helidon.config.Config;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static com.hedera.block.server.Constants.BLOCKNODE_STORAGE_ROOT_PATH_KEY;
+
+/**
+ * The FileSystemBlockStorage class implements the BlockStorage interface to store blocks to the filesystem.
+ */
+public class FileSystemBlockStorage implements BlockStorage<BlockStreamServiceGrpcProto.Block> {
```

Comment:
I don't know that there's a high likelihood we'll ever use direct/simple file system storage.
We need to compress block items to have sane storage cost, we need to manage individual items, not just whole blocks, and we need very low latency as well as low cost for large volume. We almost certainly will want something like a MerkleDb, Parquet, or some form of low-latency distributed storage.

Reply:
I understand.  Those decisions are still being hashed out.  I expect this will change but the original requirements from stand-up discussions was that we would use something that just writes to the filesystem.